datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/ChecksumEnforcingInputStream.java

```diff
@@ -76,7 +76,7 @@ public int read(byte[] b, int off, int len) throws IOException {
       endToEndChecksumHandler.update(b, off, i);
     } else {
       // no more payload to read. compute checksum and verify
-      if (!expectedChecksum.equalsIgnoreCase(endToEndChecksumHandler.hash())) {
+      if (!expectedChecksum.equals(endToEndChecksumHandler.hash())) {
```

Comment:
One QQ,
In this case, client is verifying the response integrity using checksum. IIUC  For older version of this library `expectedChecksum` variable will be in integer (long) format (received as part of response header) but `endToEndChecksumHandler.hash()` will still be returning the checksum in hex representation.

Will it be a breaking change for older version of this library or Am I missing something ?

Reply:
It looks like the backend always used or switched to integer format a long time ago. So currently `expectedChecksum` is always in integer format for any client library version. Before this PR `endToEndChecksumHandler.hash()` returns hex format so the feature is currently broken and this PR fixes that.

If we enable the `GOOGLE_CLOUD_DATASTORE_HTTP_ENABLE_E2E_CHECKSUM` with the current client all requests fail because of checksum mismatch (because of different format). But with this PR it works.

Reply:
Thanks @nimf for the clarification. It makes sense.

I think what we are missing is a separate integration test with `GOOGLE_CLOUD_DATASTORE_HTTP_ENABLE_E2E_CHECKSUM` variable set to `true` so that we can catch these issues sooner, but that can be covered as part of separate PR, I have created #1189  for that. For now it's LGTM.

cc: @kolea2 .

Reply:
Thanks! Yes, I'll add it in a separate PR.